### PR TITLE
Minor: Better ruby style & remove unnecessary short circuit check

### PR DESCRIPTION
**Pull request by apurvis**
Just noticed this was redundant; `randomize` is a private method that is never called if `rand_factor.zero?`

i also made a couple v. small style tweaks; if you don't like them i can change them back but i thought they made it slightly more readable.

### CODE REVIEW

File: lib/retriable/exponential_backoff.rb [hunk 1]
```diff
@@ -28,15 +28,16 @@ def intervals
         [base_interval * multiplier**iteration, max_interval].min
       end
 
-      return intervals if rand_factor.zero?
-
-      intervals.map { |i| randomize(i) }
+      if rand_factor.zero?
```
Comment (kamui) [inline lib/retriable/exponential_backoff.rb]: Removing the interval check in randomize makes sense, but in this change, I prefer my guard/return/fall-through style rather than if/else. Can you just revert this change, it looks good otherwise.

Comment (apurvis) [reply lib/retriable/exponential_backoff.rb]: no problem

File: lib/retriable.rb [hunk 2]
```diff
@@ -37,6 +37,7 @@ def retriable(opts = {})
     on_retry          = local_config.on_retry
     sleep_disabled    = local_config.sleep_disabled
 
+    exception_list = on.is_a?(Hash) ? on.keys : on
```
Comment (apurvis) [inline lib/retriable.rb]: (i moved this because i felt like it was hanging out in an awkward place way underneath all the other instantiation variables)
